### ISSUE TITLE
Integrate IIIF viewer

### ISSUE THREAD

**Issue by scherztc** [uclibs/scholar_uc#30]

**Comment by jamesvanmil** [comment]
We made it fairly far with Riiif/OpenSeaDragon work, but ran into to an error which isn't obviously related to either gem.

OpenSeaDragon requires some json formatted info, which Riiif provides via: https://github.com/curationexperts/riiif/blob/master/app/controllers/riiif/images_controller.rb#L20-L23

This calls an info_service method, which is configured per: https://github.com/curationexperts/riiif/blob/master/README.md#integration-with-hydrafedora

```ruby
Riiif::Image.info_service = lambda do |id, file|
  resp = get_solr_response_for_doc_id id
  doc = resp.first['response']['docs'].first
  { height: doc[HEIGHT_SOLR_FIELD], width: doc[WIDTH_SOLR_FIELD] }
end
```

This configuration yields the following error:

`undefined local variable or method `blacklight_solr' for main:Object`

`get_solr_response_for_doc_id` calls `blacklight_solr`, but it's not clear whats going wrong.

**Comment by scherztc** [comment]
I am going to move forward with installing Grunt on larry and see if that helps create the json/info page.

**Comment by scherztc** [comment]
I have worked through an implementation of riiif/openseadragon in Scholar@UC.  This implementation involves some Gemfile, initializer, and view changes in the application (https://github.com/uclibs/scholar_uc/commit/773d6bdf6596a53d9317d94fcd04cc197de59749) and the indexing of two new fields (height and width) in the fork (https://github.com/uclibs/curate_fork/commit/b0baf0ab46564298aaa290051ac52159164977d1), but does NOT require any changes to our hosted environment.

Take a look at it on http://larry.libraries.uc.edu:3006/

**Comment by scherztc** [comment]
Deployed to QA:  https://scholar-qa.uc.edu

**Comment by scherztc** [comment]
Install instructions for Redhat and lloris:  https://github.com/loris-imageserver/loris/blob/development/doc/redhat-install.md

**Comment by scherztc** [comment]
I attempted to move through the Redhat instructions, but ran into a blocker at the "Install Image Libraries" point.  Some of the links were broken and realistically, I am out of my depth with installation and configuration of various dependencies and services in Redhat.

**Comment by scherztc** [comment]
We have loris and a branch of scholar running on larry:  http://larry.libraries.uc.edu:3005/

We need to upload a few different file types and see how it responds.

**Comment by scherztc** [comment]
Merge the OpenSeaDragon branch.

**Comment by scherztc** [comment]
Thanks.  It is deployed to DEV if you want to run it through the ringers.  http://scholar-dev.uc.edu

I am looking into why the GIF’s don’t work.  (animated or otherwise).

Thomas


From: James Van Mil
Reply-To: uclibs/scholar_uc
Date: Tuesday, February 9, 2016 at 3:35 PM
To: uclibs/scholar_uc
Cc: Thomas Scherz
Subject: Re: [scholar_uc] Integrate IIIF viewer (#30)


Closed #30<https://github.com/uclibs/scholar_uc/issues/30>.
